action.yml

```diff
@@ -0,0 +1,232 @@
+            ARCH="aarch64"
+            ;;
+          *)
+            echo "Unsupported architecture: $ARCH"
+            exit 1
+            ;;
+        esac
+        
+        # Construct target triple
+        if [ "$OS" = "linux" ]; then
+          # Detect if we're on musl or glibc
+          # Check for musl library files or use ldd as fallback
+          if [ -f /lib/ld-musl-x86_64.so.1 ] || [ -f /lib/ld-musl-aarch64.so.1 ] || (ldd --version 2>&1 | grep -qi musl); then
+            TARGET="${ARCH}-unknown-linux-musl"
+          else
+            TARGET="${ARCH}-unknown-linux-gnu"
+          fi
+        elif [ "$OS" = "apple" ]; then
+          TARGET="${ARCH}-apple-darwin"
+        elif [ "$OS" = "windows" ]; then
+          TARGET="${ARCH}-pc-windows-msvc"
+        fi
+        
+        echo "os=$OS" >> $GITHUB_OUTPUT
+        echo "arch=$ARCH" >> $GITHUB_OUTPUT
+        echo "target=$TARGET" >> $GITHUB_OUTPUT
+        echo "Detected platform: $TARGET"
+
+    - name: Resolve wasixcc version
+      id: resolve-version
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+      run: |
+        VERSION="${{ inputs.version }}"
+        if [ "$VERSION" = "latest" ]; then
+          echo "Fetching latest release version..."
+          VERSION=$(curl -s -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/wasix-org/wasixcc/releases/latest | grep '"tag_name"' | sed -E 's/.*"([^"]+)".*/\1/')
+          if [ -z "$VERSION" ]; then
+            echo "Failed to fetch latest version"
+            exit 1
+          fi
+          echo "Latest version: $VERSION"
+        fi
+        echo "version=$VERSION" >> $GITHUB_OUTPUT
+
+    - name: Download wasixcc binary
+      id: download
+      shell: bash
+      run: |
+        VERSION="${{ steps.resolve-version.outputs.version }}"
+        TARGET="${{ steps.platform.outputs.target }}"
+        DOWNLOAD_URL="https://github.com/wasix-org/wasixcc/releases/download/${VERSION}/wasixcc-${TARGET}.tar.gz"
+        
+        echo "Downloading wasixcc from: $DOWNLOAD_URL"
+        curl -L -o wasixcc.tar.gz "$DOWNLOAD_URL"
+        
+        if [ ! -f wasixcc.tar.gz ]; then
+          echo "Failed to download wasixcc"
+          exit 1
+        fi
+        
+        echo "Downloaded wasixcc successfully"
+
+    - name: Install wasixcc
+      id: install-wasixcc
+      shell: bash
+      run: |
+        echo "Extracting wasixcc..."
+        tar -xzf wasixcc.tar.gz
+        rm wasixcc.tar.gz
+        
+        # Create installation directory
+        INSTALL_DIR="$HOME/.wasixcc/bin"
+        mkdir -p "$INSTALL_DIR"
+        
+        # Move binary to installation directory
+        if [ "${{ steps.platform.outputs.os }}" = "windows" ]; then
+          mv wasixcc.exe "$INSTALL_DIR/"
+          WASIXCC_PATH="$INSTALL_DIR/wasixcc.exe"
+        else
+          mv wasixcc "$INSTALL_DIR/"
+          chmod +x "$INSTALL_DIR/wasixcc"
+          WASIXCC_PATH="$INSTALL_DIR/wasixcc"
+        fi
```

Comment:
Dont have the real wasixcc executable in INSTALL_DIR, put it directly in .wasixcc. If its in INSTALL_DIR then --install-executables might overwrite it